File: orchestrator/web/src/types/artifact.ts
export interface Artifact {
  id: number
  packageId: string
  packageName: string
  version: string
  installerFile: string
  uploadedAt: string
}

export interface BulkImportResult {
  imported: { packageId: string; version: string; installerFile: string }[]
  failed: { file: string; reason: string }[]
}

File: orchestrator/backend/Models/Artifact.cs
namespace Orchestrator.Models;

public class Artifact
{
    public int Id { get; set; }
    public string PackageId { get; set; } = string.Empty;
    public string PackageName { get; set; } = string.Empty;
    public string Version { get; set; } = string.Empty;
    public string InstallerFile { get; set; } = string.Empty;
    public string ManifestPath { get; set; } = string.Empty;
    public string BinaryPath { get; set; } = string.Empty;
    public DateTime UploadedAt { get; set; } = DateTime.UtcNow;
}
